sw/device/lib/testing/test_framework/ottf_console.c

```diff
@@ -0,0 +1,167 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/test_framework/ottf_console.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/dif/dif_uart.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/irq.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_isrs.h"
+#include "sw/device/lib/testing/test_framework/ottf_test_config.h"
+
+// TODO: make this toplevel agnostic.
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/**
+ * OTTF console configuration parameters.
+ */
+enum {
+  /**
+   * Flow control parameters.
+   */
+  kFlowControlLowWatermark = 4,   // bytes
+  kFlowControlHighWatermark = 8,  // bytes
+  kFlowControlRxWatermark = kDifUartWatermarkByte8,
+  /**
+   * HART PLIC Target.
+   */
+  kPlicTarget = kTopEarlgreyPlicTargetIbex0,
+};
+
+// Potential DIF handles for OTTF console communication.
+static dif_uart_t ottf_console_uart;
+
+// The `flow_control_state` and `flow_control_irqs` variables are shared between
+// the interrupt service handler and user code.
+static volatile ottf_console_flow_control_t flow_control_state;
+static volatile uint32_t flow_control_irqs;
+
+void *ottf_console_get() {
+  switch (kOttfTestConfig.console.type) {
+    default:
+      return &ottf_console_uart;
+  }
+}
+
+void ottf_console_init(void) {
+  // Initialize/Configure the console device.
+  uintptr_t base_addr = kOttfTestConfig.console.ip_base_addr;
```

Comment:
I don't like the name `ip_base_addr` - My first thought was "so we support TCP/IP now?"  I think `base_addr` is sufficient.